core_backend/app/dashboard/models.py

```diff
@@ -114,30 +165,50 @@ async def get_timeseries(
 
 
 async def get_top_content(
-    user_id: int, asession: AsyncSession, top_n: int
-) -> List[TopContent]:
-    """
-    Retrieve most frequently shared content
+    *, user_id: int, asession: AsyncSession, top_n: int, exclude_archived: bool
+) -> list[TopContent]:
+    """Retrieve most frequently shared content.
+
+    Parameters
+    ----------
+    user_id
+        The ID of the user to retrieve the top content for.
+    asession
+        `AsyncSession` object for database transactions.
+    top_n
+        The number of top content to retrieve.
+    exclude_archived
+        Specifies whether to exclude archived content.
+
+    Returns
+    -------
+    list[TopContent]
+        List of most frequently shared content.
     """
+
     statement = (
         select(
             ContentDB.content_title,
             ContentDB.query_count,
             ContentDB.positive_votes,
             ContentDB.negative_votes,
             ContentDB.updated_datetime_utc,
+            ContentDB.is_archived,
         )
         .order_by(ContentDB.query_count.desc())
         .where(ContentDB.user_id == user_id)
```

Comment:
If `exclude_archived` is `True`, we need to add another where statement to filter those out right?

Reply:
Per our discussion, the `exclude_archived` parameter is actually not necessary here.